common/bitvec/src/lib.rs

```diff
@@ -75,6 +79,29 @@ impl BitVec {
         let bucket_pos = pos as usize - (bucket * BUCKET_SIZE);
         (self.inner[bucket] & (0x01 << bucket_pos)) != 0
     }
+
+    // TODO(kostas): Remove after applying it to multi-sig.
+    #[allow(dead_code)]
+    /// Returns the number of set bits.
+    pub fn count_ones(&self) -> u32 {
+        self.inner.iter().map(|a| a.count_ones()).sum()
+    }
+
+    // TODO(kostas): Remove after applying it to multi-sig.
+    #[allow(dead_code)]
+    /// Returns the index of the last set bit or -1 if no bits are set.
+    pub fn last_set_bit(&self) -> i32 {
+        for (i, byte) in self.inner.iter().rev().enumerate() {
+            if *byte != 0 {
+                for j in (0..8).rev() {
+                    if *byte & (1 << j) != 0 {
+                        return ((self.inner.len() - 1 - i) * 8 + j as usize) as i32;
```

Comment:
Might be good to use the `trailing_zeros` function: https://doc.rust-lang.org/std/primitive.u8.html#method.trailing_zeros